pkg/ddevapp/apptypes_test.go

```diff
@@ -106,11 +106,59 @@ func TestConfigOverrideAction(t *testing.T) {
 		app.PHPVersion = newVersion
 		err = app.WriteConfig()
 		assert.NoError(err)
-		err = app.ConfigFileOverrideAction()
+		err = app.ConfigFileOverrideAction(false)
 		assert.NoError(err)
 		// But with a config that has been written with a specified version, the version should be untouched by
-		// app.ConfigFileOverrideAction()
 		assert.EqualValues(app.PHPVersion, newVersion)
 	}
+}
+
+// TestConfigOverrideActionOnExistingConfig tests that the ConfigOverride action is properly applied, even if the config
+// existed when the override flag is enabled.
+func TestConfigOverrideActionOnExistingConfig(t *testing.T) {
+	assert := asrt.New(t)
+	origDir, _ := os.Getwd()
+
+	// This will only work for those project types defining configOverrideAction and altering php version
+	appTypes := map[string]string{
+		nodeps.AppTypeCakePHP: nodeps.PHP83,
+		nodeps.AppTypeDrupal6: nodeps.PHP56,
+		nodeps.AppTypeDrupal7: nodeps.PHP82,
+		// For AppTypeDrupal we can't guess a version without a working installation.
```

Comment:
Now that you added e.g. `cmd/ddev/cmd/testdata/TestConfigUpdate/drupal11-composer/` we could improve test coverage here.

Reply:
By a lot. Maybe we should throw this test away and move everything over there.